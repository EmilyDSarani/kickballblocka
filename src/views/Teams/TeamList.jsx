import React from 'react'
import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getTeams } from '../../services/teams'

export default function TeamList() {
    const [teams, setTeams] = useState([]);

    useEffect(()=>{
        getTeams().then((sponse)=>setTeams(sponse));
    },[]);

    return (
        <>
        <div>
            <h1>Team Page</h1>
            <p>New to the Club?</p>
            <Link to ='/teams/newteam'>Register Your Team </Link>
        </div>
        <ul>
            {teams.map((team)=>{
                return(
                    <li key={team.id}>
                        <Link to={`/teams/${team.id}`}>
                            {team.name}
                        </Link>
                    </li>
                );
            })}
        </ul>
        </>
    )
}
