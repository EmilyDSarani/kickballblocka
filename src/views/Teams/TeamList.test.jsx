import { render } from '@testing-library/react';
import TeamList from './TeamList';

it('should render a team list', async () => {
    
    const { container } = render(<TeamList />);
    expect(container).toMatchSnapshot();
})

//I would like some help going over how to test for a list to be rendering? I am not sure what was really needed to test here. 