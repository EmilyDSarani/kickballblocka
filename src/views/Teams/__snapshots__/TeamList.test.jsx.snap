// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a team 1`] = `
<div>
  <div>
    <h1>
      Team Page
    </h1>
  </div>
  <ul />
</div>
`;

exports[`should render a team list 1`] = `
<div>
  <div>
    <h1>
      Team Page
    </h1>
  </div>
  <ul />
</div>
`;
